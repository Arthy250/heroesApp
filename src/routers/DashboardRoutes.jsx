import React from 'react'
import { Navbar } from '../components/ui/NavBar';
import { MarvelScreen } from '../components/marvel/MarvelScreen';
import { HeroesScreen } from '../components/heroes/HeroesScreen';
import { DcScreen } from '../components/dc/DcScreen';
import { Switch, Route, Redirect } from "react-router-dom";
import { SearchScreen } from '../components/search/SearchScreen';


export const DashboardRoutes = () => {
  return (
    <>
      <Navbar/>

      <div className="container mt-2">
        <Switch>

          <Route exact path="/marvel" component={MarvelScreen}></Route>
          <Route exact path="/heroe/:heroeId" component={HeroesScreen}></Route>
          <Route exact path="/dc" component={DcScreen}></Route>
          <Route exact path="/search" component={SearchScreen}></Route>

          <Redirect to="/marvel" />
          
        </Switch>
      </div>
    </>
  )
}
