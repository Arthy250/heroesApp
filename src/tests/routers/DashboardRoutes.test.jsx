import React from 'react';
import {mount} from 'enzyme';
import {MemoryRouter} from 'react-router-dom';
import { DashboardRoutes } from '../../routers/DashboardRoutes';
import { AuthContext } from '../../auth/AuthContext';

describe('Pruebas en <DashboardRoutes/>', () => {
  
  const contextValue = {
    dispatch: jest.fn(),
    user: {
      logged: true,
      name: 'Arturo'
    }
  }

  test('Debe mostrarse correctamente', () => {
    
    const wrapper = mount(
      <AuthContext.Provider value={contextValue}>
        <MemoryRouter>
          <DashboardRoutes/>
        </MemoryRouter>
      </AuthContext.Provider>
    );

    expect(wrapper).toMatchSnapshot();

  })
  
})
