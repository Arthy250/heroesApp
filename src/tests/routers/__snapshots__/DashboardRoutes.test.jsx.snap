// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <DashboardRoutes/> Debe mostrarse correctamente 1`] = `
Array [
  <nav
    className="navbar navbar-expand-sm navbar-dark bg-dark"
  >
    <a
      className="navbar-brand"
      href="/"
      onClick={[Function]}
    >
      Asociaciones
    </a>
    <div
      className="navbar-collapse"
    >
      <div
        className="navbar-nav"
      >
        <a
          aria-current="page"
          className="nav-item nav-link active"
          href="/marvel"
          onClick={[Function]}
          style={Object {}}
        >
          Marvel
        </a>
        <a
          aria-current={null}
          className="nav-item nav-link"
          href="/dc"
          onClick={[Function]}
        >
          DC
        </a>
        <a
          aria-current={null}
          className="nav-item nav-link"
          href="/search"
          onClick={[Function]}
        >
          SEARCH
        </a>
      </div>
    </div>
    <div
      className="navbar-collapse collapse w-100 order-3 dual-collapse2"
    >
      <ul
        className="navbar-nav ml-auto"
      >
        <span
          className="nav-item nav-link text-info"
        >
           
          Arturo
           
        </span>
        <button
          className="btn nav-item nav-link"
          onClick={[Function]}
        >
          Logout
        </button>
      </ul>
    </div>
  </nav>,
  <div
    className="container mt-2"
  >
    Array [
      <h1>
        Heroes Marvel
      </h1>,
      <hr />,
      <div
        className="card-columns animate__animated animate__fadeIn"
      >
        <div
          className="card ms-3"
          style={
            Object {
              "maxWidth": 540,
            }
          }
        >
          <div
            className="row no-gutters"
          >
            <div
              className="col-md-4"
            >
              <img
                alt="Spider Man"
                className="card-img"
                src="./assets/heroes/marvel-spider.jpg"
              />
            </div>
            <div
              className="col-md-8"
            >
              <div
                className="card-body"
              >
                <div
                  className="card-title"
                >
                  Spider Man
                </div>
                <p
                  className="card-text"
                >
                  Peter Parker
                </p>
                <div
                  className="card-text"
                >
                  <p
                    className="text-muted"
                  >
                     
                    Amazing Fantasy #15
                     
                  </p>
                </div>
                <a
                  href="/heroe/marvel-spider"
                  onClick={[Function]}
                >
                  Mas...
                </a>
              </div>
            </div>
          </div>
        </div>
        <div
          className="card ms-3"
          style={
            Object {
              "maxWidth": 540,
            }
          }
        >
          <div
            className="row no-gutters"
          >
            <div
              className="col-md-4"
            >
              <img
                alt="Captain America"
                className="card-img"
                src="./assets/heroes/marvel-captain.jpg"
              />
            </div>
            <div
              className="col-md-8"
            >
              <div
                className="card-body"
              >
                <div
                  className="card-title"
                >
                  Captain America
                </div>
                <p
                  className="card-text"
                >
                  Steve Rogers
                </p>
                <div
                  className="card-text"
                >
                  <p
                    className="text-muted"
                  >
                     
                    Captain America Comics #1
                     
                  </p>
                </div>
                <a
                  href="/heroe/marvel-captain"
                  onClick={[Function]}
                >
                  Mas...
                </a>
              </div>
            </div>
          </div>
        </div>
        <div
          className="card ms-3"
          style={
            Object {
              "maxWidth": 540,
            }
          }
        >
          <div
            className="row no-gutters"
          >
            <div
              className="col-md-4"
            >
              <img
                alt="Iron Man"
                className="card-img"
                src="./assets/heroes/marvel-iron.jpg"
              />
            </div>
            <div
              className="col-md-8"
            >
              <div
                className="card-body"
              >
                <div
                  className="card-title"
                >
                  Iron Man
                </div>
                <p
                  className="card-text"
                >
                  Tony Stark
                </p>
                <div
                  className="card-text"
                >
                  <p
                    className="text-muted"
                  >
                     
                    Tales of Suspense #39
                     
                  </p>
                </div>
                <a
                  href="/heroe/marvel-iron"
                  onClick={[Function]}
                >
                  Mas...
                </a>
              </div>
            </div>
          </div>
        </div>
        <div
          className="card ms-3"
          style={
            Object {
              "maxWidth": 540,
            }
          }
        >
          <div
            className="row no-gutters"
          >
            <div
              className="col-md-4"
            >
              <img
                alt="Thor"
                className="card-img"
                src="./assets/heroes/marvel-thor.jpg"
              />
            </div>
            <div
              className="col-md-8"
            >
              <div
                className="card-body"
              >
                <div
                  className="card-title"
                >
                  Thor
                </div>
                <p
                  className="card-text"
                >
                  Thor Odinson
                </p>
                <div
                  className="card-text"
                >
                  <p
                    className="text-muted"
                  >
                     
                    Journey into Myster #83
                     
                  </p>
                </div>
                <a
                  href="/heroe/marvel-thor"
                  onClick={[Function]}
                >
                  Mas...
                </a>
              </div>
            </div>
          </div>
        </div>
        <div
          className="card ms-3"
          style={
            Object {
              "maxWidth": 540,
            }
          }
        >
          <div
            className="row no-gutters"
          >
            <div
              className="col-md-4"
            >
              <img
                alt="Hulk"
                className="card-img"
                src="./assets/heroes/marvel-hulk.jpg"
              />
            </div>
            <div
              className="col-md-8"
            >
              <div
                className="card-body"
              >
                <div
                  className="card-title"
                >
                  Hulk
                </div>
                <p
                  className="card-text"
                >
                  Bruce Banner
                </p>
                <div
                  className="card-text"
                >
                  <p
                    className="text-muted"
                  >
                     
                    The Incredible Hulk #1
                     
                  </p>
                </div>
                <a
                  href="/heroe/marvel-hulk"
                  onClick={[Function]}
                >
                  Mas...
                </a>
              </div>
            </div>
          </div>
        </div>
        <div
          className="card ms-3"
          style={
            Object {
              "maxWidth": 540,
            }
          }
        >
          <div
            className="row no-gutters"
          >
            <div
              className="col-md-4"
            >
              <img
                alt="Wolverine"
                className="card-img"
                src="./assets/heroes/marvel-wolverine.jpg"
              />
            </div>
            <div
              className="col-md-8"
            >
              <div
                className="card-body"
              >
                <div
                  className="card-title"
                >
                  Wolverine
                </div>
                <p
                  className="card-text"
                >
                  James Howlett
                </p>
                <div
                  className="card-text"
                >
                  <p
                    className="text-muted"
                  >
                     
                    The Incredible Hulk #180
                     
                  </p>
                </div>
                <a
                  href="/heroe/marvel-wolverine"
                  onClick={[Function]}
                >
                  Mas...
                </a>
              </div>
            </div>
          </div>
        </div>
        <div
          className="card ms-3"
          style={
            Object {
              "maxWidth": 540,
            }
          }
        >
          <div
            className="row no-gutters"
          >
            <div
              className="col-md-4"
            >
              <img
                alt="Daredevil"
                className="card-img"
                src="./assets/heroes/marvel-daredevil.jpg"
              />
            </div>
            <div
              className="col-md-8"
            >
              <div
                className="card-body"
              >
                <div
                  className="card-title"
                >
                  Daredevil
                </div>
                <p
                  className="card-text"
                >
                  Matthew Michael Murdock
                </p>
                <div
                  className="card-text"
                >
                  <p
                    className="text-muted"
                  >
                     
                    Daredevil #1
                     
                  </p>
                </div>
                <a
                  href="/heroe/marvel-daredevil"
                  onClick={[Function]}
                >
                  Mas...
                </a>
              </div>
            </div>
          </div>
        </div>
        <div
          className="card ms-3"
          style={
            Object {
              "maxWidth": 540,
            }
          }
        >
          <div
            className="row no-gutters"
          >
            <div
              className="col-md-4"
            >
              <img
                alt="Hawkeye"
                className="card-img"
                src="./assets/heroes/marvel-hawkeye.jpg"
              />
            </div>
            <div
              className="col-md-8"
            >
              <div
                className="card-body"
              >
                <div
                  className="card-title"
                >
                  Hawkeye
                </div>
                <p
                  className="card-text"
                >
                  Clinton Francis Barton
                </p>
                <div
                  className="card-text"
                >
                  <p
                    className="text-muted"
                  >
                     
                    Tales of Suspense #57
                     
                  </p>
                </div>
                <a
                  href="/heroe/marvel-hawkeye"
                  onClick={[Function]}
                >
                  Mas...
                </a>
              </div>
            </div>
          </div>
        </div>
        <div
          className="card ms-3"
          style={
            Object {
              "maxWidth": 540,
            }
          }
        >
          <div
            className="row no-gutters"
          >
            <div
              className="col-md-4"
            >
              <img
                alt="Cyclops"
                className="card-img"
                src="./assets/heroes/marvel-cyclops.jpg"
              />
            </div>
            <div
              className="col-md-8"
            >
              <div
                className="card-body"
              >
                <div
                  className="card-title"
                >
                  Cyclops
                </div>
                <p
                  className="card-text"
                >
                  Scott Summers
                </p>
                <div
                  className="card-text"
                >
                  <p
                    className="text-muted"
                  >
                     
                    X-Men #1
                     
                  </p>
                </div>
                <a
                  href="/heroe/marvel-cyclops"
                  onClick={[Function]}
                >
                  Mas...
                </a>
              </div>
            </div>
          </div>
        </div>
        <div
          className="card ms-3"
          style={
            Object {
              "maxWidth": 540,
            }
          }
        >
          <div
            className="row no-gutters"
          >
            <div
              className="col-md-4"
            >
              <img
                alt="Silver Surfer"
                className="card-img"
                src="./assets/heroes/marvel-silver.jpg"
              />
            </div>
            <div
              className="col-md-8"
            >
              <div
                className="card-body"
              >
                <div
                  className="card-title"
                >
                  Silver Surfer
                </div>
                <p
                  className="card-text"
                >
                  Norrin Radd
                </p>
                <div
                  className="card-text"
                >
                  <p
                    className="text-muted"
                  >
                     
                    The Fantastic Four #48
                     
                  </p>
                </div>
                <a
                  href="/heroe/marvel-silver"
                  onClick={[Function]}
                >
                  Mas...
                </a>
              </div>
            </div>
          </div>
        </div>
      </div>,
    ]
  </div>,
]
`;
